ostelco-ios-client/Models/MyInfoDetails.swift

```diff
@@ -17,78 +17,101 @@ struct MyInfoAddress{
     let floor: String?
     let building: String?
 
-    static func fromJSON(_ json: JSON) -> MyInfoAddress {
-        return MyInfoAddress(
-            country: json["country"].string,
-            unit: json["unit"].string,
-            street: json["street"].string,
-            block: json["block"].string,
-            postal: json["postal"].string,
-            floor: json["floor"].string,
-            building: json["building"].string
-        )
-    }
     func getAddressLine1() -> String {
         var addressLine1: String = ""
-        if let block = block {
+        if let block = self.block {
             addressLine1 = "\(block) "
         }
-        if let street = street {
+        if let street = self.street {
             addressLine1 += "\(street) "
         }
         return addressLine1
     }
+    
     func getAddressLine2() -> String {
         var addressLine2: String = ""
-        if let postal = postal {
+        if let postal = self.postal {
             addressLine2 = "\(postal) "
         }
-        if let country = country {
+        if let country = self.country {
             addressLine2 += "\(country) "
         }
         return addressLine2
     }
 }
 
-struct MyInfoDetails {
-    let name: String
-    let dob: String
-    let email: String
+struct MyInfoDetails: Codable {
+    private let _name: MyInfoRequiredValue
+    var name: String {
+        return _name.value
+    }
+    
+    private let _dob: MyInfoRequiredValue
+    var dob: String {
+        return _dob.value
+    }
+    private let _email: MyInfoRequiredValue
+    var email: String {
+        return _email.value
+    }
+    
+    private let _sex: MyInfoOptionalValue?
+    var sex: String? {
+        return _sex?.value
+    }
+    
+    private let _residentialStatus: MyInfoOptionalValue?
+    var residentialStatus: String? {
+        return _residentialStatus?.value
+    }
+    
+    private let _nationality: MyInfoOptionalValue?
+    var nationality: String? {
+        return _nationality?.value
+    }
+    
     let address: MyInfoAddress
-    let sex: String?
-    let residentialStatus: String?
-    let nationality: String?
-    let mobileNumber: String?
+    let mobileNumber: MyInfoMobileNumber?
```

Comment:
`address` and `mobileNumber` are directly exposed since there's some underlying details we may need to preserve there.